app/Http/Controllers/StudentsCouncil/AnonymousQuestionController.php

```diff
@@ -37,36 +35,34 @@ public function __construct()
      * the list of questions
      * and the option to add new ones.
      */
-    public function indexSemesters()
+    public function index()
     {
         $this->authorize('administer', AnswerSheet::class);
 
-        return view('student-council.anonymous-questions.index_semesters');
+        return view('student-council.anonymous-questions.index');
     }
 
     /**
      * Returns the 'new question' page.
      */
-    public function create(Semester $semester)
+    public function create()
     {
         $this->authorize('administer', AnswerSheet::class);
 
-        if ($semester->isClosed()) {
+        if (!$this->isActive()) {
             abort(403, "tried to add a question to a closed semester");
```

Comment:
@viktorcsimma 
actually you don't need this. Just check if $this->periodicEvent() is not null. 
Then you can create questions for the current period, regardless if it has been opened or closed. 
Just make sure that the users update the period and the semester in the semester evaluation page beforehand (or show the currently selected semester at the question form).

Reply:
I see... this seems to be fine.